plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/TestKafkaIntegrationPushDown.java

```diff
@@ -101,70 +100,60 @@ public void testPartitionPushDown()
-        DistributedQueryRunner queryRunner = getDistributedQueryRunner();
-        // ">= startTime" insure including index 2, "< endTime"  insure excluding index 4;
-        String sql = format(
-                "SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'",
-                topicNameCreateTime,
-                recordMessage.getStartTime(),
-                recordMessage.getEndTime());
-
-        // timestamp_upper_bound_force_push_down_enabled default as false.
-        assertEventually(() -> {
-            MaterializedResultWithQueryId queryResult = queryRunner.executeWithQueryId(getSession(), sql);
-            assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
-                    .isEqualTo(998);
-        });
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp < timestamp '%s'", topicNameCreateTime, recordMessage.getEndTime()), 1000, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp <= timestamp '%s'", topicNameCreateTime, recordMessage.getEndTime()), 1000, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp > timestamp '%s'", topicNameCreateTime, recordMessage.getStartTime()), 997, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s'", topicNameCreateTime, recordMessage.getStartTime()), 998, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp between timestamp '%s' and timestamp '%s'", topicNameCreateTime, recordMessage.getStartTime(), recordMessage.getEndTime()), 998, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'", topicNameCreateTime, recordMessage.getStartTime(), recordMessage.getEndTime()), 998, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp = timestamp '%s'", topicNameCreateTime, recordMessage.getStartTime()), 998, session);
 
         // timestamp_upper_bound_force_push_down_enabled set as true.
-        assertEventually(() -> {
-            // timestamp_upper_bound_force_push_down_enabled set as true.
-            Session sessionWithUpperBoundPushDownEnabled = Session.builder(getSession())
-                    .setSystemProperty("kafka.timestamp_upper_bound_force_push_down_enabled", "true")
-                    .build();
-
-            MaterializedResultWithQueryId queryResult = queryRunner.executeWithQueryId(sessionWithUpperBoundPushDownEnabled, sql);
-            assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
-                    .isEqualTo(2);
-        });
+        session = Session.builder(getSession())
```

Comment:
Let's rename to `sessionWithUpperBoundPushDownEnabled`

Reply:
I used sessionWithUpperBoundPushDownEnabled instead of session